augur/filters/exclude_by_date.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+import functools
+
+from augur.utils import ambiguous_date_to_date_range
+from augur.filters.base_filter import BaseFilter
+
+
+def is_enabled(args):
+    return args.min_date or args.max_date
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "--min-date", type=str, help="minimal cutoff for numerical date"
+    )
+    parser.add_argument(
+        "--max-date", type=str, help="maximal cutoff for numerical date"
+    )
+
+
+def init(context):
+    # TODO do not like reaching all the way back to the SequenceFile
+    if "date" not in context.sequence_file.metadata_columns:
+        raise "Date cutoff specified, but `date` is not in the metadata columns"
+
+    # TODO fromisoformat is a behavior change! might as well introduce some predictability to the input. backwards compatibility?
+    if context.args.min_date:
+        min_date = datetime.date.fromisoformat(context.args.min_date)
+    else:
+        min_date = None
+    if context.args.max_date:
+        max_date = datetime.date.fromisoformat(context.args.max_date)
+    else:
+        max_date = None
+    return ExcludeByDate(min_date=min_date, max_date=max_date)
+
+
+class ExcludeByDate(BaseFilter):
+    def __init__(self, *, min_date=None, max_date=None):
```

Comment:
why the `*`?

Reply:
A single `*` parameter name designates all following parameters as keyword-only (https://www.python.org/dev/peps/pep-3102/).

Since the two parameters here do not have an obvious meaning (well, maybe you can assume two ends of a range with the larger on the right (if your primary language is left-to-right...?)), keyword-only arguments keep things unambiguous.

Reply:
Huh, TIL